aiohttp/web_fileresponse.py

```diff
@@ -122,29 +131,31 @@ async def _precondition_failed(
         return await super().prepare(request)
 
     def _get_file_path_stat_and_gzip(
-        self, check_for_gzipped_file: bool
-    ) -> Tuple[pathlib.Path, os.stat_result, bool]:
-        """Return the file path, stat result, and gzip status.
+        self, accept_encoding: str
+    ) -> Tuple[pathlib.Path, os.stat_result, Optional[str]]:
+        """Return the file path, stat result, and possible compression type.
 
         This method should be called from a thread executor
         since it calls os.stat which may block.
         """
         filepath = self._path
-        if check_for_gzipped_file:
-            gzip_path = filepath.with_name(filepath.name + ".gz")
-            try:
-                return gzip_path, gzip_path.stat(), True
-            except OSError:
-                # Fall through and try the non-gzipped file
-                pass
-
-        return filepath, filepath.stat(), False
+        for encoding, extension in ENCODING_EXTENSION.items():
+            if encoding in accept_encoding:
+                compressed_path = filepath.with_name(filepath.name + extension)
+                try:
+                    return compressed_path, compressed_path.stat(), encoding
+                except OSError:
+                    # Try the next extension
+                    pass
+
+        # Fallback to the uncompressed file
+        return filepath, filepath.stat(), None
 
     async def prepare(self, request: "BaseRequest") -> Optional[AbstractStreamWriter]:
         loop = asyncio.get_event_loop()
-        check_for_gzipped_file = "gzip" in request.headers.get(hdrs.ACCEPT_ENCODING, "")
-        filepath, st, gzip = await loop.run_in_executor(
-            None, self._get_file_path_stat_and_gzip, check_for_gzipped_file
+        accept_encoding = request.headers.get(hdrs.ACCEPT_ENCODING, "").lower()
```

Comment:
Since we are going to make these case-insensitive (that looks like what is expected according to https://www.rfc-editor.org/rfc/rfc9110#name-content-codings), we should also have a test for this

Reply:
> All content codings are case-insensitive and ought to be registered within the "HTTP Content Coding Registry", as described in [Section 16.6](https://www.rfc-editor.org/rfc/rfc9110#content.coding.extensibility)

This looks like this should be a separate bug fix PR so we could backport the `lower()` change to 3.9.x

Reply:
If we can include links to the relevant part of the spec where we do such code changes (as I've done throughout http_parser.py, for example), that would help maintainability. You can get a link to a specific paragraph.

Reply:
> It looks like this is a bugfix to make the check "case-insensitive" and comply with the above-quoted section of the RFC
> 
> For this case, we should do a separate bug fix PR to make this so we can backport the `lower()` change to 3.9.x, along with some tests to make sure various cases work.

Yes, I essentially included that as a fix, and to be consistent with the equivalent check for dynamic compression.  I concur it should be a separate PR though so I'll remove it here.

Reply:
It would be good to merge the separate PR first since this one is likely to backport cleanly to 3.10 from `master` if we rebase it on top of that change

Reply:
Opened #8127 to resolve this.

Reply:
Thanks